packages/cli/src/commands/rule-author.ts

```
@@ -47,46 +61,42 @@ export async function ruleAuthorCommand(opts: {
-          // with one, surface the teardown loudly beside it — never replace it.
-          if (primaryErr === undefined) throw teardownErr;
-          console.warn(
-            `[RuleAuthor] WARNING: sandbox teardown failed after a primary error (root kept at ${sandbox.root}): ${teardownErr instanceof Error ? teardownErr.message : String(teardownErr)}`,
-          );
-        }
+        removeAuthorSandbox({ lcDir, root: sandbox.root, safeExec });
+      } catch (teardownErr) {
+        // A throw in `finally` SHADOWS the in-flight error (GCA #2293 round 4):
+        // with no primary in flight the teardown failure IS the error (throw);
+        // with one, surface the teardown loudly beside it — never replace it.
+        if (primaryErr === undefined) throw teardownErr;
```

Biome (2.5.1)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)